cocotb/scheduler.py

```diff
@@ -424,6 +424,14 @@ def react(self, trigger, depth=0):
                     if pending.primed:
                         pending.unprime()
                     del self._trigger2coros[pending]
+                # can't unprime trigger because other coroutines are waiting,
+                # but we should remove all coroutines being woken from the list
```

Comment:
I think you might just want to remove the `if` above and do this unconditionally. So:
```
self._trigger2coros[pending] = [
    coro
    for coro in self._trigger2coros[pending]
    if coro not in scheduling_set
]
if not self._trigger2coros[pending]:
    if pending.primed:
        pending.unprime()
    del self._trigger2coros[pending]
```